lib/ansible/modules/network/basics/uri.py

```diff
@@ -403,8 +403,12 @@ def main():
 
     # Grab all the http headers. Need this hack since passing multi-values is
     # currently a bit ugly. (e.g. headers='{"Content-Type":"application/json"}')
+    warnings = set()
     for key, value in six.iteritems(module.params):
         if key.startswith("HEADER_"):
+            warnings.add('Supplying headers via HEADER_* is deprecated and '
```

Comment:
Ok, I'll just leave this for right now.  If I can get around to adding a `deprecations` facility I will, otherwise, I'll just wait until you get to it (since you have an idea of how you would like it to work), and then see about updating this PR accordingly.